ext/BesselsEnzymeCoreExt.jl

```diff
@@ -35,4 +35,17 @@ module BesselsEnzymeCoreExt
       Duplicated(ls, dls)
   end
 
+  # This is fixing a straight bug in Enzyme.
+  function EnzymeRules.forward(func::Const{typeof(sinpi)}, 
```

Comment:
Any way we can get this fixed in enzyme itself?

Reply:
Also, this should use `sincospi`

Reply:
I poked over at https://github.com/EnzymeAD/Enzyme.jl/issues/443 
I don’t think I know exactly how to solve that though

Reply:
As it turns out this is not the only problem. Something else in the generic power series is broken for `Enzyme` but not `ForwardDiff`. Leaving a summary comment below, one moment.

Reply:
When you have time could post a specific example where this is broken? I will try to figure out what line is causing the issue even when separating out the sinpi. 

These issues though are especially annoying......

Reply:
Wait I'm kinda looking at this more carefully. Is the issue in Enzyme really localized to just `sinpi` and not `cospi`? It wouldn't make much sense to me for enzyme to give correct results for cospi but not sinpi in this situation? So if we manually define the derivative it would seem that cospi would also give incorrect derivatives?

Reply:
Ironically sincospi using Enzyme should be fine. I'm adding a pr for sinpi/cospi now which hopefully will be available in a few days.

Reply:
Also, though I'm solving in a different way from this PR (internal to Enzyme proper rather than Enzyme.jl custom rule), rules like this are welcome as PR's to Enzyme.jl

Reply:
Perfect! Thanks for looking at this. I'll change it over here once that is available.

Reply:
It occurs to me to mention here for people looking at this PR in the future that the problem _was_ just the `sinpi`, but that I didn't understand how to properly write `EnzymeRules` and just needed to propagate the `x.dval` in the derivative part of the returned `Duplicated` object. I didn't include tests for power series accuracy here because that will probably be a little bit of a project to get the last few digits, but once I fixed my custom rule that worked fine.

@wsmoses, would you like me to make a PR with this rule in the meantime? If it is fixed and will be available in the next release, maybe not point unless you would make a more immediate release that has the custom rule. I'd be happy to try and make that PR if you want, but I understand if it isn't the most useful.

Sorry that this thread looks on a skim like Enzyme problems but was actually "Chris doesn't know how to write custom rules" problems.

Reply:
See https://github.com/EnzymeAD/Enzyme/pull/1216. Hopefully that fixes this issue here and we can remove that part in the future.

P.s. I have the general `besselk` working now locally so hopefully we can get that merged soon and can test the general derivative cases.